src/core/shared_library.cc

```diff
@@ -60,16 +72,55 @@ OpenLibraryHandle(const std::string& path, void** handle)
 
     return Status(
         Status::Code::NOT_FOUND,
-        "unable to set dll path for backend library: " + errstr);
+        "unable to set dll path " + path + ": " + errstr);
   }
+#endif
+
+  return Status::Success;
+}
+
+Status
+SharedLibrary::ResetLibraryDirectory()
+{
+#ifdef _WIN32
```

Comment:
Why do we need to reset library path for windows?

Reply:
We don't want the path we set for one backend to be used for anything else. More generally we could probably try to read what the original path was and restore it. But that is not necessary for the use cases we care about.